Backport thread priority enhancement for 4.x

nealef opened the pull request:
Retrofit of the thread priority enhancement to 3.12. I checked with Miguel and as we're not expecting new RPMs to be generated, just having this code in the repo will be okay.

akoeplinger commented:
Can you please add the #ifdef I added in https://github.com/mono/mono/commit/7d4021505251772d42d3180c356681db3a51b014 ?

nealef commented:
Added.

Can you please add the #ifdef I added in 7d40215
<https://github.com/mono/mono/commit/7d4021505251772d42d3180c356681db3a51b01
4>  ?
+IBQ
Reply to this email directly or view it on GitHub
<https://github.com/mono/mono/pull/2565#issuecomment-179446626> .
